app/scripts/modules/amazon/src/function/configure/Network.tsx

```diff
@@ -0,0 +1,246 @@
+          if (provider === 'aws') {
+            forOwn(sgByRegion, function(groups, region) {
+              if (region === values.region) {
+                groups.forEach(function(group) {
+                  if (group.vpcId === values.vpcId) {
+                    sgOptions.push({ value: group.id, label: group.name });
+                  }
+                });
+              }
+            });
+          }
+        });
+      }
+    });
+    return sgOptions;
+  };
+
+  public render() {
+    const { vpcOptions, subnets, securityGroups } = this.state;
+    const { values } = this.props.formik;
+    const subnetOptions = (subnets || []).map(this.toSubnetOption);
+    const sgOptions = securityGroups ? this.getSecurityGroupsByVpc(securityGroups) : [];
+    return (
+      <div className="form-group">
+        <div className="col-md-11">
+          <div className="sp-margin-m-bottom">
+            {values.credentials && (
+              <FormikFormField
+                name="vpcId"
+                label="VPC Id"
+                help={<HelpField id="aws.function.vpc.id" />}
+                fastField={false}
+                input={props => (
+                  <ReactSelectInput
+                    inputClassName="cloudfoundry-react-select"
+                    {...props}
+                    stringOptions={vpcOptions
+                      .filter((v: IVpc) => v.account === values.credentials)
+                      .map((v: IVpc) => v.id)}
+                    clearable={true}
+                    value={values.vpcId}
+                  />
+                )}
+                onChange={this.handleVpcChange}
+                required={false}
+              />
+            )}
+          </div>
+          <div className="form-group">
+            <div className="col-md-4 sm-label-right">
+              <b>Subnets </b>
+              <HelpField id="aws.function.subnet" />
+            </div>
+            <div className="col-md-7">
+              {subnetOptions.length === 0 && (
+                <div className="form-control-static">No subnets found in the selected account/region/VPC</div>
+              )}
+              {values.vpcId ? (
+                <TetheredSelect
+                  multi={true}
+                  options={subnetOptions}
+                  value={values.subnetIds}
+                  onChange={this.handleSubnetUpdate}
+                />
+              ) : null}
+            </div>
+          </div>
+          <div className="form-group">
+            <div className="col-md-4 sm-label-right">
+              <b>Security Groups </b>
+              <HelpField id="aws.function.subnet" />
+            </div>
+            <div className="col-md-7">
+              {sgOptions.length === 0 && (
+                <div className="form-control-static">No security groups found in the selected account/region/VPC</div>
+              )}
+              {values.credentials && values.credentials !== 'test' && values.vpcId ? (
```

Comment:
This check for `credentials !== 'test'` seems wrong.

Reply:
Fixed. change will appear in the next push.